amida-tech/amida-auth-microservice
# Feature/js external auth

#### What's this PR do?
Adds a `/facebook` endpoint to the auth route for logging in via facebook. User's can `GET` this endpoint to log in using Facebook OAuth. This returns a JWT than can be used just like a JWT from a local login. The Facebook login also generates a `User` object in the database, with a field indicating that it is managed by an external provider. It has no password management functionality, and the `username` is just the Facebook-returned email.

#### Related JIRA tickets:
SER-11

#### How should this be manually tested?
Submit a `GET` to `/api/v0/auth/facebook` and log in with your facebook account. The returned JWT should work as normal.
